### PR TITLE
metrics to distinguish why repair packets are dropped

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Repair metric `dropped_requests` does not distinguish between packets proactively dropped during batching and those dropped due to bandwidth exhaustion.

#### Summary of Changes
Distinguish reason for packet drops in separate metrics.

Fixes #

### CODE REVIEW

File: core/src/serve_repair.rs [hunk 1]
```diff
@@ -151,7 +151,8 @@ impl RequestResponse for AncestorHashesRepairType {
 #[derive(Default)]
 struct ServeRepairStats {
     total_requests: usize,
-    dropped_requests: usize,
```
Comment (jbiseda) [inline core/src/serve_repair.rs:154]: @behzadnouri I saw you asked for this to be tracked for pager-duty. I'll follow up to have this updated. Do you know if there are other consumers which may be tracking `dropped_requests`?

Comment (behzadnouri) [reply core/src/serve_repair.rs:154]: I don't see it in any of grafana dashboards, so probably not.

@c3pko is this one of the metrics in your spreadsheet?

File: core/src/serve_repair.rs [hunk 2]
```diff
@@ -468,7 +469,7 @@ impl ServeRepair {
             }
         }
 
-        stats.dropped_requests += dropped_requests;
```
Comment (behzadnouri) [inline core/src/serve_repair.rs:471]: hmmm, now i am wondering if we are unnecessary dropping requests here.

File: core/src/serve_repair.rs [hunk 3]
```diff
@@ -151,7 +151,8 @@ impl RequestResponse for AncestorHashesRepairType {
 #[derive(Default)]
 struct ServeRepairStats {
     total_requests: usize,
-    dropped_requests: usize,
+    dropped_requests_bandwidth: usize,
+    dropped_requests_batching: usize,
```
Comment (behzadnouri) [inline core/src/serve_repair.rs:155]: `_batching` is a bit confusing to me.
maybe `_load_shed' or `_rate_limit`.

also `_outbound_bandwidth` to be more explicit.

**Review by behzadnouri — APPROVED** [review]
lgtm

File: core/src/serve_repair.rs (hunk 2 above)
Comment (jbiseda) [reply core/src/serve_repair.rs:471]: yeah, now we'll find out how many are being dropped here.

**Pull request merged**